dev-docs/show-prebid-ads-on-amp-pages.md

```diff
@@ -0,0 +1,349 @@
+
+```javascript
+    function sendAdToCreative(ev) {
+      var key = ev.message ? 'message' : 'data';
+      var data = {};
+      try {
+        data = JSON.parse(ev[key]);
+      } catch (e) {
+        // Do nothing.  No ad found.
+      }
+      if (data.adId) {
+          // AMP ads a `context` object to `window`s and that is used to find the
+          // `master` iframe where Prebid is loaded
+        var adObject = window.context.master.pbjs._bidsReceived.find(function (bid) {
+          return bid.adId === data.adId;
+        });
+
+        var ad = adObject.ad;
+        var adUrl = adObject.adUrl;
+        var message = JSON.stringify({
+          message: 'Prebid creative sent: ' + data.adId,
+          ad: ad,
+          adUrl: adUrl
+        });
+        ev.source.postMessage(message, '*');
+      }
+    }
+```
+
+
+#### 5. Load Prebid.js (header)
+
+The function `loadPrebidJS()` shown below is what's called by the AMP `draw3p` function in this page's body (see next step), which is the integration point between AMP and content in third-party iframes (which is what this page is).
+
+For an example showing how `draw3p` is used, see [this AMP test](https://dfp-amp-testing-1185.appspot.com/amp_tests/dfp-3p-iframe.html).
+
+For more information about `draw3p`, see:
+
++ [Its implementation](https://github.com/ampproject/amphtml/blob/e5501a30adf15c8fef049729f5e0e3137dbb18ca/3p/integration.js#L252)
+
++ [The issue that led to its implementation](https://github.com/ampproject/amphtml/issues/1210)
+
+```javascript
+    // Helper that runs inside `loadPrebidJS()`
+    function initAdserver() {
+      var i;
+      var adCalls = window.context.master.adCalls;
+      var adCallsLength = adCalls.length;
+      for (i = 0; i < adCallsLength; i++) {
+        adCalls.pop()();
+      }
+    }
+
+    /** wrap the rest of the setup in a function that will be called by the
+     * AMP `draw3p` hook see
+     * this example for more info:
+     * https://dfp-amp-testing-1185.appspot.com/amp_tests/dfp-3p-iframe.html
+     * */
+     function loadPrebidJS() {
+        pbjs.que.push(function () {
+            pbjs.logging = true;
+            pbjs.addAdUnits(adUnits);
+
+            pbjs.requestBids({
+                bidsBackHandler: function (bidResponses) {
+                    initAdserver();
+                    console.log('bidsBackHandler responses: ', bidResponses);
+                },
+                timeout: requestBidsDuration
+            });
+        });
+    }
+```
+
+
+#### 6. Implement `draw3p`, the integration point with AMP (body)
+
+In the page's body, implement the `draw3p` function as shown below.  Don't forget to add the arrays showing the ad networks used and the domains allowed to load this file in an iframe.
+
+`draw3p` is the integration point between AMP and content in third-party iframes (which is what this page is).
```

Comment:
Thanks, great catch!  Removed in https://github.com/prebid/prebid.github.io/pull/135/commits/15b01ec1c7ea80e219969cb9d7c25497de1b2990